cmd/fluxctl/automate_cmd.go

```diff
@@ -35,23 +33,12 @@ func (opts *serviceAutomateOpts) Command() *cobra.Command {
 }
 
 func (opts *serviceAutomateOpts) RunE(cmd *cobra.Command, args []string) error {
-	if len(args) > 0 {
-		return errorWantedNoArgs
+	policyOpts := &servicePolicyOpts{
+		rootOpts:   opts.rootOpts,
+		outputOpts: opts.outputOpts,
+		service:    opts.service,
+		cause:      opts.cause,
+		automate:   true,
 	}
-	if opts.service == "" {
-		return newUsageError("-s, --service is required")
-	}
-
-	serviceID, err := flux.ParseServiceID(opts.service)
-	if err != nil {
-		return err
-	}
-
-	jobID, err := opts.API.UpdatePolicies(noInstanceID, policy.Updates{
-		serviceID: policy.Update{Add: policy.Set{policy.Automated: "true"}},
-	}, opts.cause)
-	if err != nil {
-		return err
-	}
-	return await(cmd.OutOrStdout(), cmd.OutOrStderr(), opts.API, jobID, false, opts.verbose)
+	return policyOpts.RunE(policyOpts.Command(), args)
```

Comment:
This makes me a bit nervous -- I don't know if I trust `.Command()` to not mutate the option fields from immediately above. Does it definitely not do that? Can we be sure?

Reply:
Good point. I'll figure it out after rebass.

Reply:
FWIW, flags are only parsed once so this approach didn't break anything. I've changed the code to just use `cmd` anyway, since a future change to `spf13/cobra` could break the CLI.